Task 11: Sort articles

sausagroll90 opened the pull request:
Users can now sort articles by votes, comments, date, author or title using a drop down menu.

pippaamy commented on __tests__/app.test.js:
```diff
@@ -153,31 +153,46 @@ describe("/api/articles/:article_id", () => {
+        .expect(200)
+        .then(({ body: { article } }) => {
+          expect(article.comment_count).toBe(2);
+        });
```
Great job doing new tests for this!

pippaamy reviewed (commented):
Great job! :)

sausagroll90 committed:
Feature: Add sort menu to home page

Users can now sort articles by votes, comments, date, author or title using a drop down menu.

nc-rob-anderson commented on src/modules/api-requests.js:
```diff
@@ -1,17 +1,13 @@
-        );
-    }
+export async function getArticles(params) {
+    console.log("params -->", params);
```
Try to remove console logs before pushing up code

nc-rob-anderson reviewed (commented):
Nice work Dom! I came across a strange bug where the articles weren't being sorted by comments correctly, but I am unable to replicate it since 🤷🏼‍♂️

<img width="622" alt="Screenshot 2024-05-09 at 09 51 35" src="https://github.com/sausagroll90/nc-news/assets/150431177/9df7daf9-1185-4d48-b208-98c36f5f3d8c">

It might be nice to allow users to sort by votes desc if your API allows this etc, not just date.
